kytos/core/auth.py

```diff
@@ -87,6 +95,76 @@ def _generate_token(cls, username, time_exp):
             algorithm=cls.encode_algorithm,
         )
 
+    def create_user(self):
+        """Create user to database"""
+        req = request.json
+        utc_now = datetime.datetime.utcnow()
+        data = {
+            "username": req["username"],
+            "email": req["email"],
+            "password": req["password"],
+            "inserted_at": utc_now,
+            "updated_at": utc_now,
+        }
+        try:
+            user = UserDoc(**data)
+        except ValueError as error:
+            raise error
+        result = self.db.users.insert_one(user.dict())
+        return result
+
+    def delete_user(self, uid):
+        """Delete user from database"""
+        result = self.db.users.delete_one({"id": uid})
+        return result
+
+    def update_user(self, uid):
+        """Update user from database"""
+        req = request.json
+        allowed = ["username", "email", "password"]
+
+        data = {}
+        for key, value in req.items():
+            if key in allowed:
+                data[key] = value
+        data.update({"updated_at": datetime.datetime.utcnow()})
+        result = self.db.users.find_one_and_update({"id": uid}, data)
+        return result
+
+    def get_user(self, uid):
+        """Return a user information from database"""
+        data = self.db.users.aggregate([
+            {
+                "$match": {"id": uid}
```

Comment:
To avoid a full collection scan, let's perform the query on `username` that you'll create an index, I think we can omit `_id` and `id` from the projections for now, but then include the `username` too, that way users don't get confused with the difference between id and username. From the user perspective their id is their username, but data-model wise we'll also need to have the `_id` that MongoDB always require, and then we can conveniently update the `username` while `_id` has some gotchas that we'll avoid